lib/SVF-FE/SymbolTableBuilder.cpp

```diff
@@ -205,9 +205,76 @@ void SymbolTableBuilder::buildMemModel(SVFModule* svfModule)
 void SymbolTableBuilder::collectNullPtrBlackholeSyms(const Value *val)
 {
     if (LLVMUtil::isNullPtrSym(val))
-        symInfo->nullPtrSyms.insert(val);
+        symInfo->getModule()->addNullPtrSyms(val);
     if (LLVMUtil::isBlackholeSym(val))
-        symInfo->blackholeSyms.insert(val);
+        symInfo->getModule()->addBlackholeSyms(val);
+}
+
+
+void SymbolTableBuilder::collectSpecialSym(const Value* val){
+    if (LLVMUtil::isPtrInDeadFunction(val))
+    {
+        symInfo->getModule()->addPtrInDeadFunction(val);
+    }
+    if (const Function *fun = SVFUtil::dyn_cast<Function>(val))
+    {
+        const SVFFunction *svfFun = symInfo->getModule()->getSVFFunction(fun);
+        if (LLVMUtil::isDeadFunction(fun))
+        {
+            symInfo->getModule()->addDeadFunction(fun);
+        } 
+        else if (LLVMUtil::functionDoesNotRet(fun))
+        {
+            symInfo->getModule()->addFunctionDoesNotRet(fun);
+        }
+        if (!isExtCall(svfFun))
+        {
+            const BasicBlock* exitBB = LLVMUtil::getFunExitBB(fun);
+            symInfo->getModule()->addFunExitBB(fun,exitBB);
+
+            for (Function::const_iterator bit = fun->begin(), ebit = fun->end(); bit != ebit; ++bit)
```

Comment:
I don't fully understand this part, I'll have to leave to @yuleisui

Reply:
add fields (DeadFunction, functionDoesNotRet and FunExitBB) in SVFFunction for these three attributes